private/buf/cmd/buf/command/push/push.go

```diff
@@ -167,7 +188,7 @@ func run(
 	if err != nil {
 		return err
 	}
-	modulePin, err := push(ctx, container, moduleIdentity, builtModule, flags)
+	modulePin, err := pushOrCreate(ctx, container, moduleIdentity, builtModule, flags)
```

Comment:
Nit also in favor of clarity: I'd call the `createRepo` func inside the `if flags.CreateVisibility != "" {` block, and in here I'd just keep doing a normal push.

Reply:
We generally do all the flag validation up front, so that the command fails fast on invalid input, and so that we take no action if the input is invalid.

Reply:
And not only that, if we're not invoking createRepo RPC every time the flag is passed (as my other comment suggested), then my comment here makes not much sense, create would need to happen only if push fails.